### PR TITLE
Add missing CodeScanning endpoints

### PR DESCRIPTION
**Pull request by jporzucek**
Fixes https://github.com/google/go-github/issues/2899

### CODE REVIEW

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:403]: Please double-check your URLs, as this one didn't work.

```suggestion
// GitHub API docs: https://docs.github.com/en/free-pro-team@latest/rest/code-scanning/code-scanning?apiVersion=2022-11-28#get-information-about-a-sarif-upload
```

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:404]:
```suggestion
func (s *CodeScanningService) GetSARIF(ctx context.Context, owner, repo, sarifID string) (*SARIFUpload, *Response, error) {
```

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:474]:
```suggestion
// DeleteAnalysis represents a successful deletion of a code scanning analysis.
```

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:490]: Please delete this line, as it will break the auto-generated godoc comment processor.

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:508]:
```suggestion
// ListCodeQLDatabases lists the CodeQL databases that are available in a repository.
```

File: github/code-scanning.go
Comment (gmlewis) [inline github/code-scanning.go:531]:
```suggestion
// GetCodeQLDatabase gets a CodeQL database for a language in a repository.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @jporzucek .
Just a few changes to make, please, mostly relating to capitalizations, then we should be ready for a second LGTM+Approval from any other contributor to this repo before merging.

**Review by gmlewis — APPROVED** [review]
Thank you, @jporzucek !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by WillAbides — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @WillAbides !
Merging.

**Pull request merged**
